src/fs/directory.rs

```diff
@@ -282,17 +282,27 @@ fn in_memory_filesystem(record: &js_sys::Object) -> Result<virtual_fs::mem_fs::F
             create_dir_all(&fs, parent)?;
         }
 
+        tracing::trace!(
+            path=%path.display(),
+            file.length=contents.len(),
+            "Adding file to directory",
+        );
         fs.insert_ro_file(&path, contents.into())
             .with_context(|| format!("Unable to write to \"{}\"", path.display()))?;
     }
 
     Ok(fs)
 }
 
+#[tracing::instrument(level = "trace", skip(fs))]
 fn create_dir_all(fs: &dyn FileSystem, path: &Path) -> Result<(), anyhow::Error> {
     let ancestors: Vec<&Path> = path.ancestors().collect();
 
     for ancestor in ancestors.into_iter().rev() {
+        if fs.read_dir(ancestor).is_ok() {
+            continue;
+        }
```

Comment:
I don't think this is properly implemented? Since you are only getting each Path separately.

For example, if you have `/a/b/c/` you should do `read_dir("/a")`, `read_dir("/a/b")` ...but this implementation is doing `read_dir("/a")`, `read_dir("/b")`

Reply:
Nope, this is correct. The `path.ancestors()` iterator yields `/a/b/c` then `/a/b` then `/a` then `/`.

Reply:
Oh, good to know!